### ISSUE TITLE
[Feature Request] Introduce hooks

### ISSUE THREAD

**Issue by Backfighter** [CyberShadow/aconfmgr#49]
Some setup actions need to perfromed after a package has been installed or after specific files have already been copied over.

For example if you want to use a specific shell (zsh) you have to install the package and copy over all necessary configuration files (.zshrc, ...). 
Then you have to set the shell as your default shell. This can only be done after the package has already been installed.

So we need a way to specify scripts that are run after aconfmgr apply.
So basically a function `PostApply` which runs given code or script file after `aconfmgr apply` has been executed.

**Comment by CyberShadow** [comment]
Any specific reason to do it this way, instead of doing it in the currently achievable way by including `/etc/passwd` in your configuration (which includes users' shells)?

**Comment by Backfighter** [comment]
Basically flexibility when working with more than one host. 

Passwd includes the shell for each user, but also a lot more like home directories and group ids. I might want the same terminal setup on all my machines but not the same users and groups. 

Using passwd forces me to also version users and groups which I don't want to do.

**Comment by CyberShadow** [comment]
> Basically flexibility when working with more than one host.
> 
> Passwd includes the shell for each user, but also a lot more like home directories and group ids. I might want the same terminal setup on all my machines but not the same users and groups.

Have a look here:

https://github.com/CyberShadow/aconfmgr/wiki/User-management

With these helpers, it's possible to break up the user configuration and make parts of it conditional.

> Using passwd forces me to also version users and groups which I don't want to do.

Be aware that the system user and group list is a critical part of the system configuration. Not including it in your configuration would mean that restoring the configuration on a new machine will likely not result in a functional system.

The problem with post-install hooks is that their effects would be opaque to aconfmgr; it wouldn't know when they need to be applied and what the effect of applying them was, so, they don't really fit in aconfmgr's data model.

If you still don't want to version all users / groups, a compromise that would better fit with the way aconfmgr works is to modify the current system's passwd file in your configuration. I.e.:

```bash
while IFS=: read -r username ppassword uid gid desc home shell
do
	if [[ "$username" == Backfighter ]]
	then
		shell=/usr/bin/zsh
	fi
	printf '%s:%s:%d:%d:%s:%s:%s\n' "$username" "$ppassword" "$uid" "$gid" "$desc" "$home" "$shell"
done < /etc/passwd > "$(CreateFile /etc/passwd)"
```
